integration/volume_copy_up_test.go

```diff
@@ -144,34 +215,46 @@ func TestVolumeOwnership(t *testing.T) {
 	assert.Equal(t, expectedContainerOutput, string(stdout))
 
 	t.Logf("Check ownership of test directory on the host")
-	volumePaths, err := getHostPathForVolumes(*criRoot, cn)
+	// volumePaths, err := getHostPathForVolumes(*criRoot, cn)
+	volumePaths, err := getContainerVolumes(t, *criRoot, cn)
 	require.NoError(t, err)
 	assert.Equal(t, len(volumePaths), 1, "expected exactly 1 volume")
 
-	output, err := getOwnership(volumePaths[0])
+	output, err := getOwnership(volumePaths[volumePath])
 	require.NoError(t, err)
 	assert.Equal(t, expectedHostOutput, output)
 }
 
-func getHostPathForVolumes(criRoot, containerID string) ([]string, error) {
-	hostPath := filepath.Join(criRoot, "containers", containerID, "volumes")
-	if _, err := os.Stat(hostPath); err != nil {
-		return nil, err
-	}
-
-	volumes, err := os.ReadDir(hostPath)
-	if err != nil {
-		return nil, err
+func getContainerVolumes(t *testing.T, criRoot, containerID string) (map[string]string, error) {
+	client, err := RawRuntimeClient()
+	require.NoError(t, err, "failed to get raw grpc runtime service client")
+	request := &v1.ContainerStatusRequest{
+		ContainerId: containerID,
+		Verbose:     true,
 	}
+	response, err := client.ContainerStatus(context.TODO(), request)
+	require.NoError(t, err)
+	ret := make(map[string]string)
 
-	if len(volumes) == 0 {
-		return []string{}, nil
-	}
+	mounts := struct {
+		RuntimeSpec struct {
+			Mounts []specs.Mount `json:"mounts"`
+		} `json:"runtimeSpec"`
+	}{}
 
-	volumePaths := make([]string, len(volumes))
-	for idx, volume := range volumes {
-		volumePaths[idx] = filepath.Join(hostPath, volume.Name())
+	info := response.Info["info"]
+	err = json.Unmarshal([]byte(info), &mounts)
+	require.NoError(t, err)
+	containerVolumesHostPath := filepath.Join(criRoot, "containers", containerID, "volumes")
+	for _, mount := range mounts.RuntimeSpec.Mounts {
+		// norm, err := getFinalPath(mount.Source)
+		// require.NoError(t, err)
```

Comment:
Nit: Leaked code. Was this WIP? I don't see `getFinalPath` in this file. (I didn't check elsewhere though.)

Reply:
Volume mount sources on Windows use the GUID volume path. Was playing around with converting it to a DOS path, but it's not worth it in any way. On windows we only have the mounts in the image, no need to do a prefix check. So we can just check for the files inside the volume, on the host and its contents.